### ISSUE TITLE
Remove $where in subscriptions queries in favour of "inverse regex" operation

### ISSUE THREAD

**Issue by fgalan** [telefonicaid/fiware-orion#3517]
After years waiting for it :) it seems that the "inverse regex" functionality is going to be included in MongoDB 4.2 (see https://jira.mongodb.org/browse/SERVER-11947). This will allow to remove the usage of $where to search for subscriptions during an entity update in the noCache mode.

Before starting to implement this issue, first MongoDB 4.2 should become in the official Orion version (currently is MongoDB 3.6). Even at that moment, this issue has low priority, as the noCache model is not the recommended one.

However, I'm writing this down in an issue to not forgetting about it :)
